impeller/renderer/backend/metal/allocator_mtl.mm

```diff
@@ -194,6 +195,11 @@ static MTLStorageMode ToMTLStorageMode(StorageMode mode,
 
   mtl_texture_desc.storageMode = ToMTLStorageMode(
       desc.storage_mode, supports_memoryless_targets_, supports_uma_);
+  if (@available(macOS 12.5, ios 15.0, *)) {
```

Comment:
TODO: make configurable, figure out if it has reasonable behavior for image textures